plugins/aws/cli_provisioner.go

```diff
@@ -25,22 +26,36 @@ func (p awsCLIProvisioner) Provision(ctx context.Context, in sdk.ProvisionInput,
 
 func stripAndReturnProfileFlag(args *[]string) (string, error) {
 	for i, arg := range *args {
+		// if `--profile` is used after `--`, it should not be interpreted as a flag
+		if arg == "--" {
+			break
+		}
+
 		if arg == "--profile" {
 			if i+1 == len(*args) {
 				return "", fmt.Errorf("--profile flag was specified without a value")
 
 			}
 			profile := (*args)[i+1]
 			// Remove --profile flag so aws cli doesn't attempt to assume role by itself
-			*args = removeProfileFlagFromArgs(i, *args)
+			*args = removeArgSequenceFromArgList(i, i+1, *args)
+			return profile, nil
+		} else if strings.Contains(arg, "--profile=") {
```

Comment:
Let's use `strings.HasPrefix()`. Otherwise `"This is a story about the --profile= flag"` is also accepted.

Reply:
Makes sense, done, thanks!